carbon-design-system/carbon-addons-iot-react
# fix(stateful-table): fix multi-sort and advanced filters cooperatively

Closes #3254 

**Summary**

- fixes a couple of bugs preventing MultiSort and AdvancedFilters to be used on a stateful table at the same time.

**Change List (commits, features, bugs, etc)**

- pass currently selected advanced filters when saving multi-sort
- only add anticipated multi-sort column if it doesn't already exist
- add e2e tests to confirm

**Acceptance Test (how to verify the PR)**

- go to the [playground story](https://deploy-preview-3268--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--playground)
- turn on MultiSort and Advanced Filters knobs
- set advanced filters
- add multi-sort columns
- confirm table shows data
- repeat same process, but add multi-sort first and then add a filter

**Regression Test (how to make sure this PR doesn't break old functionality)**

- confirm multi-sort and advanced filters still function as expected



**Things to look for during review**
